direct/src/showbase/Audio3DManager.py

```diff
@@ -262,6 +262,15 @@ def update(self, task=None):
             tracked_sound = 0
             while tracked_sound < len(self.sound_dict[known_object]):
                 sound = self.sound_dict[known_object][tracked_sound]
+
+                if known_object.isEmpty():
+
+                    # remove the object from the sound dict since the object's
+                    # node path is empty; and it's no longer being used.
+                    del self.sound_dict[known_object]
+
+                    return Task.cont
```

Comment:
I don't think it should break out of updating all of the sounds if only one of them happens to have been removed.  It should instead continue the next iteration of the for loop.